compiler/resolution/functionResolution.cpp

```diff
@@ -2934,6 +2935,12 @@ static bool resolveBuiltinCastCall(CallExpr* call)
       call->baseExpr->remove();
       call->primitive = primitives[PRIM_CAST];
 
+      // swap the order of the two arguments
+      //  have: val, type
+      //  want: type, val
+      valueSe->remove();
+      targetTypeSe->insertAfter(valueSe);
```

Comment:
Is this still necessary?  It seems like it should be handled by earlier code

Reply:
Yes, this is still necessary, because PRIM_CAST still uses the opposite argument order from the way we write it in Chapel. PRIM_CAST puts the target type first and then the value. It used to be, with `_cast`, that the function call matched this order, but now it is reversed compared to the operator `:`.

Reply:
Doesn't that imply that PRIM_CAST should be updated?

Reply:
We could update it but I didn't want to do so in this effort. For one thing, if the compiler is generating C, the current order in `PRIM_CAST` is more natural.